cookie_monster/static/cookie_monster/js/base.js

```diff
@@ -1,86 +1,115 @@
-var cookieValues = [];
+(function() {
+    var BANNER_CLASSNAME = 'cookie-monster-banner';
+    var COOKIE_IDENTIFIER = window.__js_base_cookie_bridge.cookie_identifier;
+    // var COOKIE_AGE = window.__js_base_cookie_bridge.cookie_age;
+    var COOKIE_AGE = 604800;
+    var BANNER_BTN_DATA_ATTR = 'data-cookiemonster-accept';
 
-// Checks if the cookie was set
-if (checkCookies() === false) {
-    cookieBannerFadeIn("cookieMakerContainer")
-}
+    /**
+     * Create facet for base classname
+     * @param {string} name
+     * @return {string}
+     */
+    function createFacet(name) {
+        return BANNER_CLASSNAME + '__' + name;
+    }
 
-// Set only one cookie
-function setCookieBase() {
-    createCookie();
-    cookieBannerFadeOut("cookieMakerContainer");
-}
+    /**
+     * Accept all available cookie groups
+     */
+    function acceptAllCookies() {
+        createCookie(COOKIE_IDENTIFIER, getAllCookies());
+        activateBanner('.' + BANNER_CLASSNAME, false);
+    }
 
-// Set a group of selected cookies
-function setSelectedCookies() {
-    getCheckedCookies();
-    createCookie();
-    cookieBannerFadeOut("cookieMakerContainer");
-}
+    /**
+     * Set a group of selected cookies
+     */
+    function acceptSelectedCookies() {
+        createCookie(COOKIE_IDENTIFIER, getCheckedCookies());
+        activateBanner('.' + BANNER_CLASSNAME, false);
+    }
 
-// Set all cookies in the group
-function setAllCookies() {
-    selectAllCookies();
-    createCookie();
-    cookieBannerFadeOut("cookieMakerContainer");
-}
+    /**
+     * Read cookies in the browser
+     * @return {boolean}
+     */
+    function hasAcceptedCookies() {
+        if (!document.cookie) {
+            return false;
+        }
 
-// Read cookies in the browser
-function checkCookies() {
-    var cookieList = (document.cookie) ? document.cookie.split(/[;\s=]+/).indexOf(window.__js_base_cookie_bridge.cookie_identifier) : [];
-    return cookieList !== -1;
-}
+        var cookies = document.cookie.split(/[;\s=]+/);
+        return cookies.indexOf(COOKIE_IDENTIFIER) !== -1;
+    }
 
-// Push the checked cookies into an array
-function getCheckedCookies() {
-    document.querySelectorAll('.cookieMakerContainerContent input[type="checkbox"]:checked').forEach(function (el) {
-        return cookieValues.push(' ' + el.value);
-    })
-}
+    /**
+     * Get selected cookies
+     * @return {string[]}
+     */
+    function getCheckedCookies() {
+        return Array.from(document.querySelectorAll('.' + BANNER_CLASSNAME + ' input[type="checkbox"]:checked')).map(function(el) {
```

Comment:
`Array.from` is not supported on IE https://developer.mozilla.org/it/docs/Web/JavaScript/Reference/Global_Objects/Array/from

Please use the good and old `Array.prototype.slice.call` instead